theseus/core/vectorizer.py

```diff
@@ -0,0 +1,301 @@
+from collections import defaultdict
```

Comment:
Need to add license block for all new .py files.

Reply:
Added.